pkg/command/reconciler.go

```diff
@@ -115,7 +124,12 @@ func (r *Reconciliation) initChildJobDetails() {
 		)
 		return
 	}
-	got, found, err := r.isChildJobAvailable()
+
+	if r.isChildJobAvailableFn != nil {
+		got, found, err = r.isChildJobAvailableFn()
+	} else {
+		got, found, err = r.isChildJobAvailable()
```

Comment:
I am rethinking the naming part. IMO getChildJob() & getChildJobFn() will be better names.
The rest of signature can remain same.